command/crypto/otp/verify.go

```diff
@@ -67,23 +73,73 @@ as "300ms", "-1.5h" or "2h45m". Valid time units are "ns", "us" (or "µs"), "ms"
 }
 
 func verifyAction(ctx *cli.Context) error {
-	args := ctx.Args()
-	secretFile := args.Get(0)
+	var (
+		secret, algStr string
+		period         uint
+		digits         int
+	)
 
+	secretFile := ctx.String("secret")
 	b, err := ioutil.ReadFile(secretFile)
 	if err != nil {
 		return errs.FileError(err, secretFile)
 	}
-	secret := string(b)
+	secret = string(b)
 	if strings.HasPrefix(secret, "otpauth://") {
-		url, err := otp.NewKeyFromURL(secret)
+		otpKey, err := otp.NewKeyFromURL(secret)
 		if err != nil {
 			return err
 		}
-		secret = url.Secret()
+
+		u, err := url.Parse(strings.TrimSpace(secret))
+		if err != nil {
+			return errors.Wrap(err, "error parsing TOTP Key URI in secret file")
+		}
+		q := u.Query()
+
+		secret = otpKey.Secret()
+
+		// period query param
+		periods := q["period"]
+		if len(periods) > 0 {
+			period64, err := strconv.ParseUint(periods[0], 10, 32)
```

Comment:
if instead of treating the query as a map you use q.Get("period") you will get the first period passed:
```go
if period := q.Get("period"); period != "") { ... }
```

Reply:
Same for other query values below.

Reply:
This seems to be resolved with the new commits.